server/src/main/java/org/opensearch/action/support/clustermanager/TransportClusterManagerNodeAction.java

```diff
@@ -252,63 +257,75 @@ protected void doStart(ClusterState clusterState) {
+                            transportService.sendRequest(
+                                clusterManagerNode,
+                                ClusterTermVersionAction.NAME,
+                                new ClusterTermVersionRequest(),
+                                new TransportResponseHandler<ClusterTermVersionResponse>() {
+                                    @Override
+                                    public void handleResponse(ClusterTermVersionResponse response) {
+                                        if (response.getTerm() == clusterState.term() && response.getVersion() == clusterState.version()) {
+                                            // same as local execute
+                                            threadPool.executor(executor)
+                                                .execute(
+                                                    ActionRunnable.wrap(
+                                                        GetDelegateForLocalExecute(clusterState),
+                                                        l -> clusterManagerOperation(task, request, clusterState, l)
+                                                    )
+                                                );
+                                        } else {
+                                            // cluster-manager has updated state
+                                            executeOnClusterManager(nodes.getClusterManagerNode(), clusterState);
+                                        }
+                                    }
+
+                                    @Override
+                                    public void handleException(TransportException exp) {
+                                        handleTransportException(nodes.getClusterManagerNode(), clusterState, exp);
+                                    }
+
+                                    @Override
+                                    public String executor() {
+                                        return ThreadPool.Names.SAME;
+                                    }
+
+                                    @Override
+                                    public ClusterTermVersionResponse read(StreamInput in) throws IOException {
+                                        return new ClusterTermVersionResponse(in);
                                     }
                                 }
-                            }
-                        );
+                            );
+                        } else {
+                            // should be always executed on cluster-manager
+                            executeOnClusterManager(nodes.getClusterManagerNode(), clusterState);
+                        }
                     }
                 }
             } catch (Exception e) {
                 listener.onFailure(e);
             }
         }
 
+        private ClusterTermVersionResponse readClusterTermResponse(StreamInput in) throws IOException {
```

Comment:
Nit: replace this method with use of `ClusterTermVersionResponse::new`

Reply:
removed this method as it was not required